docs/resources/service_account.md

```diff
@@ -0,0 +1,93 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "temporalcloud_service_account Resource - terraform-provider-temporalcloud"
+subcategory: ""
+description: |-
+  Provisions a Temporal Cloud Service Account.
+---
+
+# temporalcloud_service_account (Resource)
+
+Provisions a Temporal Cloud Service Account.
+
+## Example Usage
+
+```terraform
+terraform {
+  required_providers {
+    temporalcloud = {
+      source = "temporalio/temporalcloud"
+    }
+  }
+}
+
+provider "temporalcloud" {
+
+}
+
+resource "temporalcloud_namespace" "namespace" {
+  name               = "terraform"
+  regions            = ["aws-us-east-1"]
+  accepted_client_ca = base64encode(file("${path.module}/ca.pem"))
+  retention_days     = 14
+}
+
+resource "temporalcloud_service_account" "global_service_account" {
+  name           = "admin"
+  account_access = "Admin"
+}
+
+resource "temporalcloud_service_account" "namespace_admin" {
+  name           = "developer"
+  account_access = "Developer"
+  namespace_accesses = [
+    {
+      namespace_id = temporalcloud_namespace.namespace.id
+      permission   = "admin"
+    }
+  ]
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `account_access` (String) The role on the account. Must be one of [admin, developer, read] (case-insensitive)
+- `name` (String) The name associated with the service account.
+
+### Optional
+
+- `namespace_accesses` (Attributes List) The list of namespace accesses. (see [below for nested schema](#nestedatt--namespace_accesses))
```

Comment:
not blocking feedback, but just so you know - we've had some issues (https://github.com/temporalio/terraform-provider-temporalcloud/issues/119, https://github.com/temporalio/terraform-provider-temporalcloud/issues/116, https://github.com/temporalio/terraform-provider-temporalcloud/issues/115) that have arisen from this tight coupling of users and namespace accesses as part of the same resource. I sketched out https://github.com/temporalio/terraform-provider-temporalcloud/pull/122 but I haven't finished the full idea yet, but my thinking here was to add another resource that represents a namespace access that can be manipulated even if the user object is referenced as a data source. Something we might also want to do here, too.